addons/sale/tests/test_controllers.py

```diff
@@ -65,6 +65,38 @@ def test_access_controller(self):
         self.assertEqual(req.status_code, 303)
 
 
+@tagged('post_install', '-at_install')
+class TestSalesControllers(BaseUsersCommon, HttpCase, SaleCommon):
+    def test_sales_portal_report(self):
+        portal_so = self.sale_order.copy()
+        portal_so.message_subscribe(self.user_portal.partner_id.ids)
+
+        portal_so._portal_ensure_token()
+        token = portal_so.access_token
+
+        self.authenticate(None, None)
+
+        req = self.url_open(
+            url='/my/orders/%s?access_token=%s&report_type=pdf' % (
+                portal_so.id,
+                token,
+            ),
```

Comment:
Little tip, you could use `get_portal_url` to simplify the test (it does handle the access token generation, and you can specify the report_type you need)